kqueen/engines/eks.py

```diff
@@ -0,0 +1,220 @@
+            # TODO: check if deprovisioning response is healthy
+        except Exception as e:
+            msg = 'Deleting cluster {} failed with the following reason:'.format(self.cluster.id)
+            logger.exception(msg)
+            return False, msg
+        return True, None
+
+    def resize(self, node_count, **kwargs):
+        """ Implement Later """
+        msg = 'Resizing cluster for Eks engine is disabled'
+        return False, msg
+
+    def get_kubeconfig(self):
+        """
+        Implementation of :func:`~kqueen.engines.base.BaseEngine.get_kubeconfig`
+        """
+        if not self.cluster.kubeconfig:
+            cluster = self.client.describe_cluster(name=self.cluster.metadata['id'])
+            kubeconfig = {}
+            if cluster['cluster']['status'] != "ACTIVE":
+                return self.cluster.kubeconfig
+            self.cluster.kubeconfig = yaml.load(open("eks-kubeconfig").read()) 
+            self.cluster.kubeconfig["clusters"][0]["cluster"] = {
+                "server": cluster['endpoint'], 
+                "certificate-authority-data": cluster['certificateAuthority']['data']
+            }
+            self.cluster.kubeconfig["users"][0]["user"]["exec"]["args"][2] = cluster['name']
+            self.cluster.save()
+        return self.cluster.kubeconfig
+
+    def cluster_get(self):
+        """
+        Implementation of :func:`~kqueen.engines.base.BaseEngine.cluster_get`
+        """
+        response = {}
+        try:
+            response = self.client.describe_cluster(self.cluster.metadata['id'])
+        except Exception as e:
+            msg = 'Fetching data from backend for cluster {} failed with the following reason:'.format(self.cluster.metadata['heat_cluster_id'])
+            logger.exception(msg)
+            return {}
+        state = STATE_MAP.get(response['cluster']['status'], config.get('CLUSTER_UNKNOWN_STATE'))
+
+        key = 'cluster-{}-{}'.format(response['cluster']['name'], response['cluster']['name'])
+        cluster = {
+            'key': key,
+            'name': response['cluster']['name'],
+            'id': response['cluster']['name'],
+            'state': state,
+            'metadata': self.cluster.metadata
+        }
+        return cluster
+
+    def cluster_list(self):
+        """Is not needed in Eks"""
+        return []
+
+    @classmethod
+    def engine_status(cls, **kwargs):
+        try:
+            aws_access_key = kwargs.get('aws_access_key', '')
+            aws_secret_key = kwargs.get('aws_secret_key', '')
+        except Exception:
+            logger.exception('{} Eks Provisioner validation failed.'.format(cls.name))
+            return config.get('PROVISIONER_ERROR_STATE')
+        client = boto3.client(
+            'eks',
+            aws_access_key_id=aws_access_key,
+            aws_secret_access_key=aws_secret_key
+        )
+        try:
+            list(client.list_clusters())
```

Comment:
can we used more lightweight request here? (since this request is sending for each cluster, even for list all clusters operation)
1) at least add maxResults parameter=1
2) or use another request maybe generate_presigned_url?

Reply:
seems, No , coz afaik only list_clusters can provide valid check on all required User IAM roles/policies and etc.. I understand that its huge request, but we are locked by small api